source/common/upstream/cds_api_impl.cc

```diff
@@ -37,6 +39,8 @@ CdsApiImpl::CdsApiImpl(const envoy::api::v2::ConfigSource& cds_config,
 }
 
 void CdsApiImpl::onConfigUpdate(const ResourceVector& resources) {
+  cm_.adsMux().pause(Config::TypeUrl::get().ClusterLoadAssignment);
```

Comment:
nit: IMO it would be clearer/safer to just have an RAII guard that does pause and then does resume on destruction.